test/unit/puppetlabs/puppetserver/environment_cache_test.clj

```diff
@@ -0,0 +1,79 @@
+  ([environment]
+    {:pre [(string? environment)]}
+    {environment
+       {"/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/manifests/init.pp" []
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/binary_file.pp"
+          [{"name" "simmons::binary_file",
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/content_file.pp"
+          [{"name" "simmons::content_file"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/custom_fact_output.pp"
+          [{"name" "simmons::custom_fact_output"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/external_fact_output.pp"
+          [{"name" "simmons::external_fact_output"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/init.pp"
+          [{"name" "simmons"
+            "params" [{"name" "studio"
+                       "default_source" "undef"}
+                      {"name" "exercises"
+                       "default_literal" ["simmons::content_file"
+                                          "simmons::source_file"
+                                          "simmons::binary_file"
+                                          "simmons::recursive_directory"
+                                          "simmons::mount_point_source_file"
+                                          "simmons::mount_point_binary_file"
+                                          "simmons::custom_fact_output"
+                                          "simmons::external_fact_output"]}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/mount_point_binary_file.pp"
+          [{"name" "simmons::mount_point_binary_file"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/mount_point_source_file.pp"
+          [{"name" "simmons::mount_point_source_file"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/recursive_directory.pp"
+          [{"name" "simmons::recursive_directory"
+            "params" [{"name" "studio"}]}]
+        "/Users/batmanandrobin/.puppet-server/2.x/code/environments/production/modules/simmons/manifests/source_file.pp"
+          [{"name" "simmons::source_file"
+            "params" [{"name" "studio"}]}]}}))
+
+(deftest environment-cache
+  (testing "Create a new empty cache"
+    (is (= {} (create-cache))))
+
+  (testing "Initializing and clearing a cache"
+    (let [cache (atom (create-cache))
```

Comment:
Not specifically, no. But I kind of wanted to illustrate intended usage as tests tend to be useful as developer documentation.